hooks/useRbfTransactionData/index.ts

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useEffect, useState } from 'react';
+import { rbf } from '../../transactions';
+import { Account, AppInfo, BtcTransactionData, SettingsNetwork, StacksNetwork, StxTransactionData } from '../../types';
+import { RbfData, calculateStxData, isBtcTransaction, sortFees } from './helpers';
+import { BitcoinEsploraApiProvider, mempoolApi } from '../../api';
+
+const useRbfTransactionData = ({
+  account,
+  transaction,
+  stacksNetwork,
+  btcNetwork,
+  esploraProvider,
+  stxAvailableBalance,
+  appInfo,
+  isLedgerAccount,
+}: {
+  account: Account | null;
+  transaction?: BtcTransactionData | StxTransactionData;
+  stacksNetwork: StacksNetwork;
+  btcNetwork: SettingsNetwork;
+  esploraProvider: BitcoinEsploraApiProvider;
+  stxAvailableBalance: string;
+  appInfo: AppInfo | null;
+  isLedgerAccount: boolean;
+}): RbfData => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [rbfData, setRbfData] = useState<RbfData>({});
+  const [errorCode, setErrorCode] = useState<'SOMETHING_WENT_WRONG' | undefined>();
+
+  const fetchStxData = useCallback(async () => {
+    if (!transaction || isBtcTransaction(transaction)) {
+      return;
+    }
+    try {
+      setIsLoading(true);
+      const calculatedData = await calculateStxData(
+        transaction,
+        btcNetwork,
+        stacksNetwork,
+        appInfo,
+        stxAvailableBalance,
+      );
+      setRbfData(calculatedData);
+    } catch (err: any) {
+      setErrorCode('SOMETHING_WENT_WRONG');
+    } finally {
+      setIsLoading(false);
+    }
+  }, [transaction, btcNetwork, stacksNetwork, appInfo, stxAvailableBalance]);
```

Comment:
I still think the stx rbf belongs in its own hook 🙂

Reply:
i think what is confusing here is the name btcNetwork, cause it's actually a settings network, which includes info for stacks and btc. we could rename or make this hook to receive network.type and network.address (stacks url)

Reply:
i also think btc rbf logic belong to a separate hook. but i also think that this hook can be two utils, it doesn't need to have logic for these reasons:

* whenever we use this hook fetchRbfData is executed if dependencies change, we kinda loose some flexibility.
* we could handle the logic of data, loading and error in react query in the clients

ignore the idea to be not use hooks, if the reasons are not valid for the current uses cases